be/src/vec/functions/function_ip.h

```diff
@@ -1159,4 +1162,136 @@ class FunctionToIP : public IFunction {
     }
 };
 
+class FunctionIPv4ToIPv6 : public IFunction {
+public:
+    static constexpr auto name = "ipv4_to_ipv6";
+    static FunctionPtr create() { return std::make_shared<FunctionIPv4ToIPv6>(); }
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 1; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeIPv6>();
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) const override {
+        const auto& ipv4_column_with_type_and_name = block.get_by_position(arguments[0]);
+        WhichDataType addr_type(ipv4_column_with_type_and_name.type);
+        const auto* ipv4_column =
+                assert_cast<const ColumnIPv4*>(ipv4_column_with_type_and_name.column.get());
```

Comment:
what if column const

Reply:
Good instructions, I will improve the handling of const